include/ignition/gazebo/detail/View.hh

```diff
@@ -280,6 +280,10 @@ template<typename ...ComponentTypeTs>
 bool View<ComponentTypeTs...>::NotifyComponentRemoval(const Entity _entity,
     const ComponentTypeId _typeId)
 {
+  // if entity is still marked as to add, remove from the view
+  if (this->RequiresComponent(_typeId))
+    this->toAddEntities.erase(_entity);
```

Comment:
How do we get into this situation? Can we add a test?

Reply:
See here: https://github.com/ignitionrobotics/ign-gazebo/pull/1187#issuecomment-965682932